components/core/src/clp_s/TimestampEntry.cpp

```diff
@@ -3,7 +3,11 @@
 #include <cmath>
 
 namespace clp_s {
-void TimestampEntry::ingest_timestamp(epochtime_t timestamp) {
+void TimestampEntry::ingest_timestamp(std::string const& key, epochtime_t timestamp) {
+    if (m_key_name.empty()) {
```

Comment:
I think its probably better to have a TimestampEntry constructor that takes in this key name, and checks for existence then create a TimestampEntry before claling ingest_timestamp and merge_range. 

Less potential for bugs, and makes the code a bit easier to follow.

Reply:
Just added a constructor that takes the key_name and moved node_ids to TimestampEntry. So write/read `TimestampEntry` become symmetric.